src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -124,13 +121,16 @@ void SotaUptaneClient::reportNetworkInfo() {
 
 Json::Value SotaUptaneClient::AssembleManifest() {
   Json::Value result;
-  Json::Value unsigned_ecu_version = pacman->getManifest(uptane_repo.getPrimaryEcuSerial());
+  installed_images.clear();
```

Comment:
If a secondary manifest verification fails, the corresponding entry in `installed_images` will be deleted at the end of this method. Is this expected behaviour? Or should it only modify the map if all verifications passed?

Reply:
Ok, great!